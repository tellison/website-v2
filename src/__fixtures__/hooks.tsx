import {
  Binary,
  ContributorApiResponse,
  Contributor,
  EventAPI,
  LatestTemurin,
  MarketplaceRelease,
  MockTemurinFeatureReleaseAPI,
  MockTemurinReleaseAPI,
  News,
  NewsResponse,
  ReleaseAsset,
  TemurinReleases,
} from '../hooks';

export const createRandomTemurinRelease = (installer): ReleaseAsset => ({
  os: 'os_mock',
  architecture: 'arch_mock',
  release_link: new URL('https://release_link_mock'),
  platform_name: 'platform_name_mock',
  release_name: 'release_name_mock',
  release_date: new Date(Date.UTC(2020, 0, 1)),
  binaries: [
    {
      type: 'type_mock',
      link: new URL('https://link_mock'),
      checksum: 'checksum_mock',
      size: 0,
      extension: 'extension_mock',
      installer_link: installer ? new URL('https://installer_link_mock') : undefined,
      installer_checksum: installer ? 'installer_checksum_mock' : undefined,
      installer_size: installer ? 0 : undefined,
      installer_extension: installer ? 'installer_extension_mock' : undefined,
    },
  ],
})

export const createRandomMarketplaceRelease = (installer, id): MarketplaceRelease => ({
  release_name: 'release_name_mock',
  vendor: 'vendor_mock',
  binary: {
    os: 'os_mock',
    architecture: 'arch_mock',
    distribution: 'distribution_mock',
    image_type: 'type_mock',
    jvm_impl: 'jvm_impl_mock',
    timestamp: new Date(Date.UTC(2020, 0, 1)),
    scm_ref: 'scm_ref_mock',
    openjdk_scm_ref: 'openjdk_scm_ref_mock',
    aqavit_results_link: new URL('https://aqavit_results_link_mock'),
    tck_affidavit_link: new URL('https://tck_affidavit_link_mock'),
    package: {
      name: 'name_mock.tar.gz',
      link: new URL('https://link_mock'),
      sha265sum: `sha265sum_mock${id}`,
      sha256sum_link: new URL('https://sha256sum_link_mock'),
      signature_link: new URL('https://signature_link_mock'),
    },
    installer: installer
      ? [
          {
            name: 'installer_name_mock.msi',
            link: new URL('https://installer_link_mock'),
            sha265sum: 'installer_sha265sum_mock',
            sha256sum_link: new URL('https://installer_sha256sum_link_mock'),
            signature_link: new URL('https://installer_signature_link_mock'),
          },
        ]
      : undefined,
  },
  version: {
    major: 0,
    minor: 0,
    security: 0,
    patch: 0,
    build: 0,
    openjdk_version: 'openjdk_version_mock',
  }
})

export const mockLatestTemurin = (installer): LatestTemurin => ({
  download_count: 0,
  id: 'id_mock',
  release_link: new URL('https://release_link_mock'),
  release_name: 'release_name_mock',
  release_type: 'release_type_mock',
  timestamp: new Date(Date.UTC(2020, 0, 1)),
  updated_at: new Date(Date.UTC(2020, 0, 1)),
  vendor: 'vendor_mock',
  version_data: {
    major: 0,
    minor: 0,
    security: 0,
    patch: 0,
    build: 0,
    openjdk_version: 'openjdk_version_mock',
  },
  binaries: [
    {
      os: 'os_mock',
      architecture: 'arch_mock',
      image_type: 'type_mock',
      heap_size: 'heap_size_mock',
      download_count: 0,
      jvm_impl: 'jvm_impl_mock',
      package: {
        name: 'name_mock.tar.gz',
        link: new URL('https://link_mock'),
        checksum: 'sha265sum_mock',
        checksum_link: new URL('https://sha256sum_link_mock'),
        metadata_link: new URL('https://metadata_link_mock'),
        signature_link: new URL('https://signature_link_mock'),
        size: 0,
        download_count: 0,
      },
      installer: installer
      ?
        {
          name: 'installer_name_mock.msi',
          link: new URL('https://installer_link_mock'),
          checksum: 'installer_sha265sum_mock',
          checksum_link: new URL('https://installer_sha256sum_link_mock'),
          metadata_link: new URL('https://installer_metadata_link_mock'),
          signature_link: new URL('https://installer_signature_link_mock'),
          size: 0,
          download_count: 0,
        }
      : undefined,
    }
  ]
})

export const mockNewsAPI = (): NewsResponse => ({
  news: [
    {
      title: 'news_title_mock',
      body: 'body_mock',
      id: 'id_mock',
      link: new URL('https://link_mock'),
      date: new Date(Date.UTC(2020, 0, 1)),
    }
  ],
  pagination: {
    page: 1,
    pagesize: 5,
    result_start: 1,
    result_end: 5,
    result_size: 5,
    total_result_size: 16
  }
});

export const mockEventsAPI = (): EventAPI => ({
  events: [
    {
      title: 'events_title_mock',
      id: 'id_mock',
      infoLink: new URL('https://link_mock'),
      date: new Date(Date.UTC(2020, 0, 1)),
    }
  ]
});

export const createMockTemurinReleaseAPI = (installer): MockTemurinReleaseAPI => ({
  release_link: new URL('https://release_link_mock'),
  release_name: 'release_name_mock',
  vendor: 'vendor_mock',
  version: {
    major: 0,
    minor: 0,
    security: 0,
    patch: 0,
    build: 0,
    openjdk_version: 'openjdk_version_mock',
  },
  binary: {
    architecture: 'arch_mock',
    project: 'project_mock',
    scm_ref: 'scm_ref_mock',
    download_count: 0,
    updated_at: new Date(Date.UTC(2020, 0, 1)),
    os: 'os_mock',
    image_type: 'jdk',
    jvm_impl: 'jvm_impl_mock',
    heap_size: 'heap_size_mock',
    package: {
      name: 'name_mock.tar.gz',
      link: new URL('https://link_mock'),
      checksum: 'checksum_mock',
      checksum_link: new URL('https://checksum_link_mock'),
      metadata_link: new URL('https://metadata_link_mock'),
      signature_link: new URL('https://signature_link_mock'),
      size: 0,
      download_count: 0,
    },
    installer: installer
      ? {
          name: 'installer_name_mock.msi',
          link: new URL('https://installer_link_mock'),
          checksum: 'installer_checksum_mock',
          checksum_link: new URL('https://installer_checksum_link_mock'),
          metadata_link: new URL('https://installer_metadata_link_mock'),
          signature_link: new URL('https://installer_signature_link_mock'),
          size: 0,
          download_count: 0,
        }
      : undefined,
  }
});

export const createMockTemurinFeatureReleaseAPI = (installer): MockTemurinFeatureReleaseAPI => ({
  id: 'id_mock',
  download_count: 0,
  release_name: 'release_name_mock',
  release_link: new URL('https://release_link_mock'),
  release_type: 'release_type_mock',
  timestamp: new Date(Date.UTC(2020, 0, 1)),
  updated_at: new Date(Date.UTC(2020, 0, 1)),
  vendor: 'vendor_mock',
  version_data: {
    major: 0,
    minor: 0,
    security: 0,
    patch: 0,
    build: 0,
    openjdk_version: 'openjdk_version_mock',
  },
  binaries: [
    {
      os: 'os_mock',
      architecture: 'arch_mock',
      image_type: 'jdk',
      package: {
        name: 'name_mock.tar.gz',
        checksum: 'checksum_mock',
        link: new URL('https://link_mock'),
        size: 0,
      },
      installer: installer
        ? {
          name: 'installer_name_mock.msi',
          checksum: 'installer_checksum_mock',
          link: new URL('https://installer_link_mock'),
          size: 0,
        }
      : undefined,
    }
  ]
});

export const createRandomTemurinReleases = (installer): TemurinReleases => ({
  release_name: 'release_name_mock',
  release_link: new URL('https://release_link_mock'),
  source_url: new URL('https://source_url_mock'),
  timestamp: new Date(Date.UTC(2020, 0, 1)),
  platforms: {
    'platform_mock': {
      assets: [
        {
          os: 'os_mock',
          architecture: 'architecture_mock',
          type: 'type_mock',
          link: new URL('https://link_mock'),
          checksum: 'checksum_mock',
          size: 0,
          extension: 'extension_mock',
          installer_link: installer ? new URL('https://installer_link_mock') : undefined,
          installer_checksum: installer ? 'installer_checksum_mock' : undefined,
          installer_size: installer ? 0 : undefined,
          installer_extension: installer ? 'installer_extension_mock' : undefined,
        },
      ],
    },
  },
})

export const createRandomContributorViewData = (): Contributor => ({
  avatarUri: 'avatarUri_mock',
  profileUri: 'profileUri_mock',
  login: 'login_mock',
  contributionsCount: 0,
  commitsListUri: 'commitsListUri_mock',
  repo: 'reponame_mock',
});

export const createRandomLatestForOSData = (): Binary => ({
  release_name: 'release_name_mock',
  link: new URL('https://link_mock'),
});

export const createRandomNewsAndEventsData = (): News => ({
  news: {
    news: [
      {
        id: 'id_mock',
        title: 'title_mock',
        body: 'body_mock',
        date: new Date(Date.UTC(2020, 0, 1)),
        link: new URL('https://link_mock'),
      },
    ],
    pagination: {
      page: 1,
      pagesize: 5,
      result_start: 1,
      result_end: 5,
      result_size: 5,
      total_result_size: 16,
    }
  },
  events: [
    {
      id: 'id_mock',
      title: 'title_mock',
      infoLink: new URL('https://infoLink_mock'),
      date: new Date(Date.UTC(2020, 0, 1)),
    },
  ]
});

export const createRandomContributorViewData1Contribution = (): Contributor => ({
  avatarUri: 'avatarUri_mock',
  profileUri: 'profileUri_mock',
  login: 'login_mock',
  contributionsCount: 1,
  commitsListUri: 'commitsListUri_mock',
  repo: 'reponame_mock',
});