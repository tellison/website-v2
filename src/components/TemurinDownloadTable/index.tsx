import * as React from "react"
import { Link, Trans, useI18next } from 'gatsby-plugin-react-i18next';
import { FaDownload } from 'react-icons/fa';
import { MdVerifiedUser } from 'react-icons/md';
import { capitalize } from '../../util/capitalize';
import { localeDate } from '../../util/localeDate';

const TemurinDownloadTable = ({results}) => {
    const { language } = useI18next();

    let source
    if (results && results.source) {
        source = results.source
    }
    return (
    <>
        {source &&
            <span><a href={source.binary.package.link}>{source.release_name} <Trans>Source Code Archive</Trans></a></span>
        }
        <table id="download-table" className="table table-bordered releases-table" style={{borderSpacing: '0 10px', borderCollapse: 'separate'}}>
            <tbody className="table-light">
            {results ? (
                results.map(
                    (pkg, i): string | JSX.Element =>
                        pkg && (
                            <tr key={i}>
                                <td className="table-secondary py-4 align-middle w-25">
                                    <a href={pkg.release_link} className="link-light">
                                        <span className="text-white">{pkg.release_name}</span>
                                    </a>
                                    <span className="text-white d-block m-2">
                                        Temurin <MdVerifiedUser data-toggle="tooltip" data-placement="bottom" title="This build is JCK certified" size={25} style={{ color: '#537FB9' }}/>
                                        <Link to='/aqavit'>
                                            <img
                                                src='/images/aqavit-icon.png'
                                                width={25}
                                                alt='AQAvit logo'
                                                data-toggle="tooltip"
                                                data-placement="bottom"
                                                title="This build is AQAvit Verified"
                                                className='img-fluid'
                                            />
                                        </Link>
                                    </span>
                                    <span className="text-white text-muted">{localeDate(pkg.release_date, language)}</span>
                                </td>
                                <td className="align-middle w-20">{capitalize(pkg.os)}</td>
                                <td className="align-middle w-20">{pkg.architecture}</td>
                                <td className="align-middle">
                                    <table className="table parent mb-0 w-auto">
                                        {pkg.binaries.map(
                                            (binary, i): string | JSX.Element =>
                                                binary && (
                                                <tbody key={i} className="table-light">
                                                    {binary.installer_link && (
                                                        <BinaryTable
                                                            checksum={binary.installer_checksum}
                                                            link={binary.installer_link}
                                                            extension={binary.installer_extension}
                                                            type={binary.type}
                                                            size={binary.installer_size}
                                                            os={capitalize(pkg.os)}
                                                            arch={pkg.architecture}
                                                            version={pkg.release_name}
                                                        />
                                                    )}
                                                    <BinaryTable
                                                        checksum={binary.checksum}
                                                        link={binary.link}
                                                        extension={binary.extension}
                                                        type={binary.type}
                                                        size={binary.size}
                                                        os={capitalize(pkg.os)}
                                                        arch={pkg.architecture}
                                                        version={pkg.release_name}
                                                    />
                                                </tbody>
                                            )
                                        )}
                                    </table>
                                </td>
                            </tr>
                        )
                )
            ) :
                <tr></tr>
            }
            </tbody>
        </table>
    </>
    );
};

export default TemurinDownloadTable;

const BinaryTable = ({ checksum, link, extension, type, size, os, arch, version }) => {
    return (
        <tr>
            <td className="align-middle text-center">
                <table><tbody>
                <tr>
                    <td>
                        {`${type} - ${size} MB`}
                    </td>
                </tr>
                <tr>
                    <td>
                        <span className="fw-light">
                            <a href=""
                                data-bs-toggle="modal"
                                data-bs-target="#checksumModal"
                                data-bs-checksum={checksum}>
                                <small><Trans>Checksum</Trans></small>
                            </a>
                        </span>
                    </td>
                </tr>

                </tbody></table>
            </td>
            <td className="align-middle">
                <Link to="/download" state={{ link: link, os: os, arch: arch, pkg_type: type, java_version: version }} className="btn btn-primary" style={{width: "6em"}}>
                    <FaDownload /> {extension}
                </Link>
            </td>
        </tr>
    )
}
