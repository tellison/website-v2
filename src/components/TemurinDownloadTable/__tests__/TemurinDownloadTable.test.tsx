import React from 'react';
import { render } from '@testing-library/react';
import { describe, expect, it } from 'vitest'
import { createRandomTemurinRelease } from '../../../__fixtures__/hooks';
import TemurinDownloadTable from '..';

let releases = [
  createRandomTemurinRelease(false),
  createRandomTemurinRelease(true),
];

describe('TemurinDownloadTable component', () => {
  it('renders correctly', () => {
    const { container } = render(
      <TemurinDownloadTable
        results={releases}
      />
    );
    expect(container).toMatchSnapshot();
  });

  it('renders correctly - with source', () => {
    releases.source = {
        release_name: 'mock-release-name',
        binary: {
          package: {
              link: 'https://mock-link',
          }
        }
    };
    const { container } = render(
      <TemurinDownloadTable
        results={releases}
      />
    );
    expect(container).toMatchSnapshot();
  });

  it('renders correctly - no data', () => {
    const { container } = render(
      <TemurinDownloadTable
        results={undefined}
      />
    );
    expect(container).toMatchSnapshot();
  });
});