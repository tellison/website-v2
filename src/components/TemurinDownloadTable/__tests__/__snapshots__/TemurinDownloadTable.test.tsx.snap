// Vitest Snapshot v1

exports[`TemurinDownloadTable component > renders correctly - no data 1`] = `
<div>
  <table
    class="table table-bordered releases-table"
    id="download-table"
    style="border-spacing: 0 10px; border-collapse: separate;"
  >
    <tbody
      class="table-light"
    >
      <tr />
    </tbody>
  </table>
</div>
`;

exports[`TemurinDownloadTable component > renders correctly - with source 1`] = `
<div>
  <span>
    <a
      href="https://mock-link"
    >
      mock-release-name
       
      Text
    </a>
  </span>
  <table
    class="table table-bordered releases-table"
    id="download-table"
    style="border-spacing: 0 10px; border-collapse: separate;"
  >
    <tbody
      class="table-light"
    >
      <tr>
        <td
          class="table-secondary py-4 align-middle w-25"
        >
          <a
            class="link-light"
            href="https://release_link_mock/"
          >
            <span
              class="text-white"
            >
              release_name_mock
            </span>
          </a>
          <span
            class="text-white d-block m-2"
          >
            Temurin 
            <svg
              data-placement="bottom"
              data-toggle="tooltip"
              fill="currentColor"
              height="25"
              stroke="currentColor"
              stroke-width="0"
              style="color: rgb(83, 127, 185);"
              viewBox="0 0 24 24"
              width="25"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title>
                This build is JCK certified
              </title>
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M12 1L3 5v6c0 5.55 3.84 10.74 9 12 5.16-1.26 9-6.45 9-12V5l-9-4zm-2 16l-4-4 1.41-1.41L10 14.17l6.59-6.59L18 9l-8 8z"
              />
            </svg>
            <a
              href="/aqavit"
            >
              <img
                alt="AQAvit logo"
                class="img-fluid"
                data-placement="bottom"
                data-toggle="tooltip"
                src="/images/aqavit-icon.png"
                title="This build is AQAvit Verified"
                width="25"
              />
            </a>
          </span>
          <span
            class="text-white text-muted"
          >
            January 1, 2020
          </span>
        </td>
        <td
          class="align-middle w-20"
        >
          Os_mock
        </td>
        <td
          class="align-middle w-20"
        >
          arch_mock
        </td>
        <td
          class="align-middle"
        >
          <table
            class="table parent mb-0 w-auto"
          >
            <tbody
              class="table-light"
            >
              <tr>
                <td
                  class="align-middle text-center"
                >
                  <table>
                    <tbody>
                      <tr>
                        <td>
                          type_mock - 0 MB
                        </td>
                      </tr>
                      <tr>
                        <td>
                          <span
                            class="fw-light"
                          >
                            <a
                              data-bs-checksum="checksum_mock"
                              data-bs-target="#checksumModal"
                              data-bs-toggle="modal"
                              href=""
                            >
                              <small>
                                Text
                              </small>
                            </a>
                          </span>
                        </td>
                      </tr>
                    </tbody>
                  </table>
                </td>
                <td
                  class="align-middle"
                >
                  <a
                    class="btn btn-primary"
                    href="/download"
                    state="[object Object]"
                    style="width: 6em;"
                  >
                    <svg
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 512 512"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M216 0h80c13.3 0 24 10.7 24 24v168h87.7c17.8 0 26.7 21.5 14.1 34.1L269.7 378.3c-7.5 7.5-19.8 7.5-27.3 0L90.1 226.1c-12.6-12.6-3.7-34.1 14.1-34.1H192V24c0-13.3 10.7-24 24-24zm296 376v112c0 13.3-10.7 24-24 24H24c-13.3 0-24-10.7-24-24V376c0-13.3 10.7-24 24-24h146.7l49 49c20.1 20.1 52.5 20.1 72.6 0l49-49H488c13.3 0 24 10.7 24 24zm-124 88c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20zm64 0c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20z"
                      />
                    </svg>
                     
                    extension_mock
                  </a>
                </td>
              </tr>
            </tbody>
          </table>
        </td>
      </tr>
      <tr>
        <td
          class="table-secondary py-4 align-middle w-25"
        >
          <a
            class="link-light"
            href="https://release_link_mock/"
          >
            <span
              class="text-white"
            >
              release_name_mock
            </span>
          </a>
          <span
            class="text-white d-block m-2"
          >
            Temurin 
            <svg
              data-placement="bottom"
              data-toggle="tooltip"
              fill="currentColor"
              height="25"
              stroke="currentColor"
              stroke-width="0"
              style="color: rgb(83, 127, 185);"
              viewBox="0 0 24 24"
              width="25"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title>
                This build is JCK certified
              </title>
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M12 1L3 5v6c0 5.55 3.84 10.74 9 12 5.16-1.26 9-6.45 9-12V5l-9-4zm-2 16l-4-4 1.41-1.41L10 14.17l6.59-6.59L18 9l-8 8z"
              />
            </svg>
            <a
              href="/aqavit"
            >
              <img
                alt="AQAvit logo"
                class="img-fluid"
                data-placement="bottom"
                data-toggle="tooltip"
                src="/images/aqavit-icon.png"
                title="This build is AQAvit Verified"
                width="25"
              />
            </a>
          </span>
          <span
            class="text-white text-muted"
          >
            January 1, 2020
          </span>
        </td>
        <td
          class="align-middle w-20"
        >
          Os_mock
        </td>
        <td
          class="align-middle w-20"
        >
          arch_mock
        </td>
        <td
          class="align-middle"
        >
          <table
            class="table parent mb-0 w-auto"
          >
            <tbody
              class="table-light"
            >
              <tr>
                <td
                  class="align-middle text-center"
                >
                  <table>
                    <tbody>
                      <tr>
                        <td>
                          type_mock - 0 MB
                        </td>
                      </tr>
                      <tr>
                        <td>
                          <span
                            class="fw-light"
                          >
                            <a
                              data-bs-checksum="installer_checksum_mock"
                              data-bs-target="#checksumModal"
                              data-bs-toggle="modal"
                              href=""
                            >
                              <small>
                                Text
                              </small>
                            </a>
                          </span>
                        </td>
                      </tr>
                    </tbody>
                  </table>
                </td>
                <td
                  class="align-middle"
                >
                  <a
                    class="btn btn-primary"
                    href="/download"
                    state="[object Object]"
                    style="width: 6em;"
                  >
                    <svg
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 512 512"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M216 0h80c13.3 0 24 10.7 24 24v168h87.7c17.8 0 26.7 21.5 14.1 34.1L269.7 378.3c-7.5 7.5-19.8 7.5-27.3 0L90.1 226.1c-12.6-12.6-3.7-34.1 14.1-34.1H192V24c0-13.3 10.7-24 24-24zm296 376v112c0 13.3-10.7 24-24 24H24c-13.3 0-24-10.7-24-24V376c0-13.3 10.7-24 24-24h146.7l49 49c20.1 20.1 52.5 20.1 72.6 0l49-49H488c13.3 0 24 10.7 24 24zm-124 88c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20zm64 0c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20z"
                      />
                    </svg>
                     
                    installer_extension_mock
                  </a>
                </td>
              </tr>
              <tr>
                <td
                  class="align-middle text-center"
                >
                  <table>
                    <tbody>
                      <tr>
                        <td>
                          type_mock - 0 MB
                        </td>
                      </tr>
                      <tr>
                        <td>
                          <span
                            class="fw-light"
                          >
                            <a
                              data-bs-checksum="checksum_mock"
                              data-bs-target="#checksumModal"
                              data-bs-toggle="modal"
                              href=""
                            >
                              <small>
                                Text
                              </small>
                            </a>
                          </span>
                        </td>
                      </tr>
                    </tbody>
                  </table>
                </td>
                <td
                  class="align-middle"
                >
                  <a
                    class="btn btn-primary"
                    href="/download"
                    state="[object Object]"
                    style="width: 6em;"
                  >
                    <svg
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 512 512"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M216 0h80c13.3 0 24 10.7 24 24v168h87.7c17.8 0 26.7 21.5 14.1 34.1L269.7 378.3c-7.5 7.5-19.8 7.5-27.3 0L90.1 226.1c-12.6-12.6-3.7-34.1 14.1-34.1H192V24c0-13.3 10.7-24 24-24zm296 376v112c0 13.3-10.7 24-24 24H24c-13.3 0-24-10.7-24-24V376c0-13.3 10.7-24 24-24h146.7l49 49c20.1 20.1 52.5 20.1 72.6 0l49-49H488c13.3 0 24 10.7 24 24zm-124 88c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20zm64 0c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20z"
                      />
                    </svg>
                     
                    extension_mock
                  </a>
                </td>
              </tr>
            </tbody>
          </table>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`TemurinDownloadTable component > renders correctly 1`] = `
<div>
  <table
    class="table table-bordered releases-table"
    id="download-table"
    style="border-spacing: 0 10px; border-collapse: separate;"
  >
    <tbody
      class="table-light"
    >
      <tr>
        <td
          class="table-secondary py-4 align-middle w-25"
        >
          <a
            class="link-light"
            href="https://release_link_mock/"
          >
            <span
              class="text-white"
            >
              release_name_mock
            </span>
          </a>
          <span
            class="text-white d-block m-2"
          >
            Temurin 
            <svg
              data-placement="bottom"
              data-toggle="tooltip"
              fill="currentColor"
              height="25"
              stroke="currentColor"
              stroke-width="0"
              style="color: rgb(83, 127, 185);"
              viewBox="0 0 24 24"
              width="25"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title>
                This build is JCK certified
              </title>
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M12 1L3 5v6c0 5.55 3.84 10.74 9 12 5.16-1.26 9-6.45 9-12V5l-9-4zm-2 16l-4-4 1.41-1.41L10 14.17l6.59-6.59L18 9l-8 8z"
              />
            </svg>
            <a
              href="/aqavit"
            >
              <img
                alt="AQAvit logo"
                class="img-fluid"
                data-placement="bottom"
                data-toggle="tooltip"
                src="/images/aqavit-icon.png"
                title="This build is AQAvit Verified"
                width="25"
              />
            </a>
          </span>
          <span
            class="text-white text-muted"
          >
            January 1, 2020
          </span>
        </td>
        <td
          class="align-middle w-20"
        >
          Os_mock
        </td>
        <td
          class="align-middle w-20"
        >
          arch_mock
        </td>
        <td
          class="align-middle"
        >
          <table
            class="table parent mb-0 w-auto"
          >
            <tbody
              class="table-light"
            >
              <tr>
                <td
                  class="align-middle text-center"
                >
                  <table>
                    <tbody>
                      <tr>
                        <td>
                          type_mock - 0 MB
                        </td>
                      </tr>
                      <tr>
                        <td>
                          <span
                            class="fw-light"
                          >
                            <a
                              data-bs-checksum="checksum_mock"
                              data-bs-target="#checksumModal"
                              data-bs-toggle="modal"
                              href=""
                            >
                              <small>
                                Text
                              </small>
                            </a>
                          </span>
                        </td>
                      </tr>
                    </tbody>
                  </table>
                </td>
                <td
                  class="align-middle"
                >
                  <a
                    class="btn btn-primary"
                    href="/download"
                    state="[object Object]"
                    style="width: 6em;"
                  >
                    <svg
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 512 512"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M216 0h80c13.3 0 24 10.7 24 24v168h87.7c17.8 0 26.7 21.5 14.1 34.1L269.7 378.3c-7.5 7.5-19.8 7.5-27.3 0L90.1 226.1c-12.6-12.6-3.7-34.1 14.1-34.1H192V24c0-13.3 10.7-24 24-24zm296 376v112c0 13.3-10.7 24-24 24H24c-13.3 0-24-10.7-24-24V376c0-13.3 10.7-24 24-24h146.7l49 49c20.1 20.1 52.5 20.1 72.6 0l49-49H488c13.3 0 24 10.7 24 24zm-124 88c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20zm64 0c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20z"
                      />
                    </svg>
                     
                    extension_mock
                  </a>
                </td>
              </tr>
            </tbody>
          </table>
        </td>
      </tr>
      <tr>
        <td
          class="table-secondary py-4 align-middle w-25"
        >
          <a
            class="link-light"
            href="https://release_link_mock/"
          >
            <span
              class="text-white"
            >
              release_name_mock
            </span>
          </a>
          <span
            class="text-white d-block m-2"
          >
            Temurin 
            <svg
              data-placement="bottom"
              data-toggle="tooltip"
              fill="currentColor"
              height="25"
              stroke="currentColor"
              stroke-width="0"
              style="color: rgb(83, 127, 185);"
              viewBox="0 0 24 24"
              width="25"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title>
                This build is JCK certified
              </title>
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M12 1L3 5v6c0 5.55 3.84 10.74 9 12 5.16-1.26 9-6.45 9-12V5l-9-4zm-2 16l-4-4 1.41-1.41L10 14.17l6.59-6.59L18 9l-8 8z"
              />
            </svg>
            <a
              href="/aqavit"
            >
              <img
                alt="AQAvit logo"
                class="img-fluid"
                data-placement="bottom"
                data-toggle="tooltip"
                src="/images/aqavit-icon.png"
                title="This build is AQAvit Verified"
                width="25"
              />
            </a>
          </span>
          <span
            class="text-white text-muted"
          >
            January 1, 2020
          </span>
        </td>
        <td
          class="align-middle w-20"
        >
          Os_mock
        </td>
        <td
          class="align-middle w-20"
        >
          arch_mock
        </td>
        <td
          class="align-middle"
        >
          <table
            class="table parent mb-0 w-auto"
          >
            <tbody
              class="table-light"
            >
              <tr>
                <td
                  class="align-middle text-center"
                >
                  <table>
                    <tbody>
                      <tr>
                        <td>
                          type_mock - 0 MB
                        </td>
                      </tr>
                      <tr>
                        <td>
                          <span
                            class="fw-light"
                          >
                            <a
                              data-bs-checksum="installer_checksum_mock"
                              data-bs-target="#checksumModal"
                              data-bs-toggle="modal"
                              href=""
                            >
                              <small>
                                Text
                              </small>
                            </a>
                          </span>
                        </td>
                      </tr>
                    </tbody>
                  </table>
                </td>
                <td
                  class="align-middle"
                >
                  <a
                    class="btn btn-primary"
                    href="/download"
                    state="[object Object]"
                    style="width: 6em;"
                  >
                    <svg
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 512 512"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M216 0h80c13.3 0 24 10.7 24 24v168h87.7c17.8 0 26.7 21.5 14.1 34.1L269.7 378.3c-7.5 7.5-19.8 7.5-27.3 0L90.1 226.1c-12.6-12.6-3.7-34.1 14.1-34.1H192V24c0-13.3 10.7-24 24-24zm296 376v112c0 13.3-10.7 24-24 24H24c-13.3 0-24-10.7-24-24V376c0-13.3 10.7-24 24-24h146.7l49 49c20.1 20.1 52.5 20.1 72.6 0l49-49H488c13.3 0 24 10.7 24 24zm-124 88c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20zm64 0c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20z"
                      />
                    </svg>
                     
                    installer_extension_mock
                  </a>
                </td>
              </tr>
              <tr>
                <td
                  class="align-middle text-center"
                >
                  <table>
                    <tbody>
                      <tr>
                        <td>
                          type_mock - 0 MB
                        </td>
                      </tr>
                      <tr>
                        <td>
                          <span
                            class="fw-light"
                          >
                            <a
                              data-bs-checksum="checksum_mock"
                              data-bs-target="#checksumModal"
                              data-bs-toggle="modal"
                              href=""
                            >
                              <small>
                                Text
                              </small>
                            </a>
                          </span>
                        </td>
                      </tr>
                    </tbody>
                  </table>
                </td>
                <td
                  class="align-middle"
                >
                  <a
                    class="btn btn-primary"
                    href="/download"
                    state="[object Object]"
                    style="width: 6em;"
                  >
                    <svg
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 512 512"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M216 0h80c13.3 0 24 10.7 24 24v168h87.7c17.8 0 26.7 21.5 14.1 34.1L269.7 378.3c-7.5 7.5-19.8 7.5-27.3 0L90.1 226.1c-12.6-12.6-3.7-34.1 14.1-34.1H192V24c0-13.3 10.7-24 24-24zm296 376v112c0 13.3-10.7 24-24 24H24c-13.3 0-24-10.7-24-24V376c0-13.3 10.7-24 24-24h146.7l49 49c20.1 20.1 52.5 20.1 72.6 0l49-49H488c13.3 0 24 10.7 24 24zm-124 88c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20zm64 0c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20z"
                      />
                    </svg>
                     
                    extension_mock
                  </a>
                </td>
              </tr>
            </tbody>
          </table>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
